ksqldb-common/src/main/java/io/confluent/ksql/util/GrammaticalJoiner.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * An object which joins pieces of text, with a separator and additional final word, allowing more
+ * grammatically correct joining, e,g. "a, b, c or d".
+ */
+public final class GrammaticalJoiner {
```

Comment:
Added have more human friendly phrasing of error messages.  If anyone knows of an existing utility to do this I'm happy to switch...